src/providers/api/Flickr.py

```diff
@@ -98,108 +102,125 @@ def create_tags_list(_data):
         return None
 
 
-def extractData(_data):
+def extract_data(_data):
     title = _data.get('title')
     creator = _data.get('ownername')
-    imageURL, height, width = get_image_url(_data)
+    image_url, height, width = get_image_url(_data)
     thumbnail = _data.get('url_s')
-    license, version = getLicense(_data.get('license', -1))
-    metaData = create_meta_data_dict(_data)
-    tagData = create_tags_list(_data)
+    license_, version = get_license(_data.get('license', -1))
+    meta_data = create_meta_data_dict(_data)
+    tags = create_tags_list(_data)
 
     if 'owner' in _data:
-        creatorURL = 'www.flickr.com/photos/{}'.format(_data['owner']).strip()
+        creator_url = 'www.flickr.com/photos/{}'.format(_data['owner']).strip()
     else:
-        creatorURL = None
+        creator_url = None
 
-    foreignID = _data.get('id')
+    foreign_id = _data.get('id')
 
-    if foreignID and creatorURL:
-        foreignURL = '{}/{}'.format(creatorURL, foreignID)
+    if foreign_id and creator_url:
+        foreign_url = '{}/{}'.format(creator_url, foreign_id)
     else:
-        foreignURL = None
+        foreign_url = None
 
 
-    return create_tsv_list_row(
-        foreign_identifier=imageURL if not foreignID else foreignID,
-        foreign_landing_url=foreignURL,
-        image_url=imageURL,
+    return em.create_tsv_list_row(
+        foreign_identifier=image_url if not foreign_id else foreign_id,
+        foreign_landing_url=foreign_url,
+        image_url=image_url,
         thumbnail=thumbnail,
         width=width,
         height=height,
-        license=license,
+        license_=license_,
         license_version=version,
         creator=creator,
-        creator_url=creatorURL,
+        creator_url=creator_url,
         title=title,
-        meta_data=metaData,
-        tags=tagData,
+        meta_data=meta_data,
+        tags=tags,
         watermarked='f',
         provider=FLICKR,
         source=FLICKR
     )
 
 
-def getMetaData(_startTs, _endTs, _license, _switchDate=False):
-    procTime    = time.time()
-    pages       = 1
-    curPage     = 1
-    numImages   = 0
-    endpoint    = 'https://api.flickr.com/services/rest/?method=flickr.photos.search'
+def construct_api_query_string(
+        start_ts,
+        end_ts,
+        license_,
+        cur_page,
+        switch_date=False):
+    date_type = 'taken' if switch_date else 'upload'
+    api_query_string = (
```

Comment:
Since we're using `requests`, could we construct the query string using a dictionary? https://2.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls

I think that would be a lot more readable.

Reply:
I agree this is the correct direction to go, but implementing this change would require either changing every script that uses `etlMods.requestContent`, as well as the logging statements in `etlMods.requestContent`, or implementing a temporary solution involving deciding whether query params were appended as a string to the input URL or not, and then choosing different logging styles based on the result.  I prefer the former route, and have created a ticket #220 to implement the change.